src/services/collection-service/collection-service-files-response.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) The Arvados Authors. All rights reserved.
+//
+// SPDX-License-Identifier: AGPL-3.0
+
+import { createCollectionFilesTree, CollectionDirectory, CollectionFile, CollectionFileType, createCollectionDirectory, createCollectionFile } from "../../models/collection-file";
+import { Tree, mapTree, getNodeChildren, getNode, TreeNode } from "../../models/tree";
+import { getTagValue } from "../../common/xml";
+
+export const parseFilesResponse = (document: Document) => {
+    const files = extractFilesData(document);
+    const tree = createCollectionFilesTree(files);
+    return sortFilesTree(tree);
+};
+
+export const sortFilesTree = (tree: Tree<CollectionDirectory | CollectionFile>) => {
+    return mapTree(node => {
+        const children = getNodeChildren(node.id)(tree)
+            .map(id => getNode(id)(tree))
+            .filter(node => node !== undefined) as TreeNode<CollectionDirectory | CollectionFile>[];
```

Comment:
why getNodeChildren is has `children` defined as `string[]`. I think we should fix tree functions instead doing nasty castings here.